From 22e7bf6ea3e8ce63564315328704478c3ff64e75 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Thu, 21 Oct 2010 14:51:00 +0200
Subject: [PATCH] qtdemux: push mode; check for and warn about non-interleaved media

---
 gst/isomp4/qtdemux.c |   50 +++++++++++++++++++++++++++++++++++++++-----------
 1 files changed, 39 insertions(+), 11 deletions(-)

diff --git a/gst/isomp4/qtdemux.c b/gst/isomp4/qtdemux.c
index 5f6b077..52a0746 100644
--- a/gst/isomp4/qtdemux.c
+++ b/gst/isomp4/qtdemux.c
@@ -3938,6 +3938,43 @@ next_entry_size (GstQTDemux * demux)
   return -1;
 }
 
+/* checks for non-interleaved a/v which does not stream properly */
+static void
+gst_qtdemux_check_interleave (GstQTDemux * demux)
+{
+  gint i, j, ni = 0;
+
+  if (demux->n_streams < 2)
+    return;
+
+  /* Figure out which stream this is packet belongs to */
+  for (i = 0; i < demux->n_streams; i++) {
+    QtDemuxStream *stream = demux->streams[i];
+
+    if (!stream || stream->n_samples <= 50)
+      continue;
+
+    if (!qtdemux_parse_samples (demux, stream, 50))
+      return;
+
+    /* naive check for number of sequential samples */
+    for (j = 0; j < 50; j++) {
+      if (stream->samples[j].offset + stream->samples[j].size !=
+          stream->samples[j + 1].offset)
+        break;
+    }
+    GST_LOG_OBJECT (demux, "stream %d has %d seq samples", i, j);
+    if (j >= 50)
+      ni++;
+  }
+
+  if (ni >= 2) {
+    GST_ELEMENT_WARNING (demux, STREAM, DEMUX,
+        (_("Several non-interleaved streams detected.  "
+                "File may not play well")), (NULL));
+  }
+}
+
 static void
 gst_qtdemux_post_progress (GstQTDemux * demux, gint num, gint denom)
 {
@@ -4114,17 +4151,8 @@ gst_qtdemux_chain (GstPad * sinkpad, GstBuffer * inbuf)
           g_node_destroy (demux->moov_node);
           demux->moov_node = NULL;
           GST_DEBUG_OBJECT (demux, "Finished parsing the header");
-        } else if (fourcc == FOURCC_moof) {
-          if (demux->got_moov && demux->fragmented) {
-            GST_DEBUG_OBJECT (demux, "Parsing [moof]");
-            if (!qtdemux_parse_moof (demux, data, demux->neededbytes,
-                    demux->offset, NULL)) {
-              ret = GST_FLOW_ERROR;
-              goto done;
-            }
-          } else {
-            GST_DEBUG_OBJECT (demux, "Discarding [moof]");
-          }
+
+          gst_qtdemux_check_interleave (demux);
         } else if (fourcc == FOURCC_ftyp) {
           GST_DEBUG_OBJECT (demux, "Parsing [ftyp]");
           qtdemux_parse_ftyp (demux, data, demux->neededbytes);
