From 37c8abcdbd98512898605cb3b2a6a9f897e10879 Mon Sep 17 00:00:00 2001
From: Mark Nauwelaerts <mark.nauwelaerts@collabora.co.uk>
Date: Fri, 11 Nov 2011 13:55:48 +0000
Subject: pulsesink: do not leak clientname when setting up property

---
diff --git a/ext/pulse/pulsesink.c b/ext/pulse/pulsesink.c
index ca769787..14c70e3 100644
--- a/ext/pulse/pulsesink.c
+++ b/ext/pulse/pulsesink.c
@@ -1867,6 +1867,7 @@ gst_pulsesink_class_init (GstPulseSinkClass * klass)
   GstBaseSinkClass *bc;
   GstBaseAudioSinkClass *gstaudiosink_class = GST_BASE_AUDIO_SINK_CLASS (klass);
   GstElementClass *gstelement_class = GST_ELEMENT_CLASS (klass);
+  gchar *clientname;
 
   gobject_class->finalize = gst_pulsesink_finalize;
   gobject_class->set_property = gst_pulsesink_set_property;
@@ -1921,12 +1922,14 @@ gst_pulsesink_class_init (GstPulseSinkClass * klass)
    *
    * Since: 0.10.25
    */
+  clientname = gst_pulse_client_name ();
   g_object_class_install_property (gobject_class,
       PROP_CLIENT,
       g_param_spec_string ("client", "Client",
-          "The PulseAudio client name to use", gst_pulse_client_name (),
+          "The PulseAudio client name to use", clientname,
           G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS |
           GST_PARAM_MUTABLE_READY));
+  g_free (clientname);
 
   /**
    * GstPulseSink:stream-properties
--
cgit v0.9.0.2-2-gbebe
